### PR TITLE
Upgrade Bootstrap

### PR DESCRIPTION
**Pull request by krishnaglick**
## Change description
Upgrading Bootstrap

Some migration docs:
https://react-bootstrap.github.io/migrating/#summary-of-breaking-changes-from-v1x
https://getbootstrap.com/docs/5.0/migration/

Screen Shot 2021-11-23 at 12 01 49 PM

There's some differences but most of them are minor. I would like to point out that Button Text is black instead of white. I haven't been able to determine precisely why, or what needs overriding to make it white. But with the UI refresh looming I feel it may not be worth the effort.

### Development

- [ ] Application changes have been tested appropriately

### Impact

- [x] Code follows company security practices and guidelines
- [x] Security impact of change has been considered
- [x] Performance impact of change has been considered
- [x] Possible migration needs considered (model migrations, config migrations, etc.)

Please explain any security, performance, migration, or other impacts if relevant:

### Code review

- [x] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached where applicable.
- [x] Relevant tags have been added to the PR (bug, enhancement, internal, etc.)

### CODE REVIEW

File: components/Subscribe.tsx [hunk 1]
```diff
@@ -72,8 +72,8 @@ export default function Subscribe({ campaign }: { campaign: string }) {
             </Form.Group>
           </Col>
         </Row>
-        <Row noGutters>
-          <Col style={{ padding: 0 }}>
+        <Row className="p-2">
```
Comment (krishnaglick) [inline components/Subscribe.tsx]: I think overall padding/margin was altered for Row/Col, so this just makes things look fine.

File: pages/_app.tsx [hunk 2]
```diff
@@ -5,6 +5,7 @@ import FloatingLayout from "../components/layouts/Floating";
 import { useEffect } from "react";
 import { useRouter } from "next/router";
 import { googleAnalyticsPageView } from "../components/GoogleAnalytics";
+import SSRProvider from "react-bootstrap/SSRProvider";
```
Comment (evantahler) [inline pages/_app.tsx:8]: neat https://react-bootstrap.github.io/getting-started/server-side-rendering/

**Review by evantahler — CHANGES_REQUESTED** [review]
Awesome! I think getting up-to-date will be really important.  I also agree that in most cases we aren't trying for a pixel-perfect match, but that the styles overall /feel/ the same.

Interesting notes:
* The max container width got bigger!  Some responsive things are bigger now

Things to fix:
* Side nav (docs) links are now underlined.  They should not be. (1) 
* Markdown (docs) in-page links are now underlines.  The should not be.
* blog follow box's form and button are no-longer left aligned (2)
* Integrations bar is wonky (3)
* I think the black button text is odd... I like the white :D 
* Meet form missing padding around the submit button (4)
* Checkbox on form not near works (5)

1.
Screen Shot 2021-11-23 at 10 14 19 AM

2.
Screen Shot 2021-11-23 at 10 13 26 AM

3.
Screen Shot 2021-11-23 at 10 14 54 AM

4.
Screen Shot 2021-11-23 at 10 16 09 AM

5.
Screen Shot 2021-11-23 at 10 16 46 AM

**Review by teallarson — COMMENTED** [review]
I'll defer to Evan here overall, but nice!

I wonder if the default text for the buttons is related to Bootstrap 5 [changing its default contrast ratio](https://getbootstrap.com/docs/5.0/migration/#color-system).  Our buttons didn't have a great contrast ratio before (though I did like them better). But I'm sure they're likely to change here soon with the site update.

Just for reference in case anyone else is interested:

before: 
Screen Shot 2021-11-23 at 1 45 37 PM
(ideal is 4.5 or greater)
 
now:
Screen Shot 2021-11-23 at 1 46 14 PM

**Comment by krishnaglick** [comment]
Regarding Button text color, it's going to be black in the new UI. Maybe worth just leaving as is? I do think it's black due to contract/a11y things.

**Comment by evantahler** [comment]
> Regarding Button text color, it's going to be black in the new UI. Maybe worth just leaving as is? I do think it's black due to contract/a11y things.

Sure, leave it back then!

**Comment by krishnaglick** [comment]
I've got everything handled, except on the Trial pages this checkbox is slightly off "center".

Screen Shot 2021-11-23 at 3 18 46 PM

**Comment by krishnaglick** [comment]
> I've got everything handled, except on the Trial pages this checkbox is slightly off "center".
> 
> Screen Shot 2021-11-23 at 3 18 46 PM

Wrapped the checkbox in a small. :| It works

**Review by evantahler — APPROVED** [review]
Nice!

The integrations links are certainly different now... but the padding is nice!
Screen Shot 2021-11-23 at 1 23 16 PM

**Comment by teallarson** [comment]
> Nice!
> 
> The integrations links are certainly different now... but the padding is nice! Screen Shot 2021-11-23 at 1 23 16 PM

The columns are off-center with the text but I don't think it's blocking

**Review by teallarson — APPROVED** [review]

**Comment by krishnaglick** [comment]
Screen Shot 2021-11-24 at 9 32 41 AM

@tealjulia @evantahler How do you feel about a really wide button?

**Comment by teallarson** [comment]
> Screen Shot 2021-11-24 at 9 32 41 AM
> 
> @tealjulia @evantahler How do you feel about a really wide button?

Mildly negative but mostly neutral.  This component has driven me nuts a few times trying to get things to line up just right!

**Comment by krishnaglick** [comment]
> > Screen Shot 2021-11-24 at 9 32 41 AM
> > @tealjulia @evantahler How do you feel about a really wide button?
> 
> Mildly negative but mostly neutral. This component has driven me nuts a few times trying to get things to line up just right!

Okay, just going to merge then. Will will be through here at some point and we can fix it then!

**Pull request merged**